File: frontend/user/lib/data/mapper/discount/discount_mapper.dart
import 'package:b2205946_duonghuuluan_luanvan/data/models/discount/discount_model.dart';
import 'package:b2205946_duonghuuluan_luanvan/domain/entity/discount/discount.dart';

class DiscountMapper {
  static Discount fromModel(DiscountModel model) {
    return Discount(
      id: model.id,
      categoryId: model.categoryId,
      name: model.name,
      description: model.description ?? "",
      percent: model.percent,
      startAt: model.startAt ?? DateTime(1970),
      endAt: model.endAt ?? DateTime(1970),
      status: model.status ?? "",
    );
  }
}

File: backend/app/infrastructure/database/mappers/discount_mapper.py
from app.domain.entities.discount_entity import DiscountEntity
from app.infrastructure.database.models.discount import Discount


class DiscountMapper:
    @staticmethod
    def to_entity(model: Discount) -> DiscountEntity:
        return DiscountEntity(
            id=model.id,
            category_id=model.category_id,
            name=model.name,
            description=model.description,
            percent=model.percent,
            status=model.status.value if hasattr(model.status, 'value') else model.status,
            start_at=model.start_at,
            end_at=model.end_at,
            created_at=model.created_at,
        )
